### PR TITLE
Update `@gear-js/api` readme

### PR DESCRIPTION
**Pull request by osipov-mit**

### CODE REVIEW

File: api/README.md [hunk 1]
```diff
@@ -71,32 +81,34 @@ result.toString(); //  - returns a string represetation of the value
 result.toU8a(); // - encodes the value as a Unit8Array
 result.toJSON(); // - converts the value to JSON
 ```
+</details>
 
 ### Getting metadata
 
-Getting metadata from program.meta.wasm
+It's possible to get the definition of program types from the `.meta.wasm` file obtained after building the program.
+Use the `getWasmMetadata` function for these purpose
```
Comment (nikitayutanov) [inline api/README.md:89]:
```suggestion
Use the `getWasmMetadata` function for this purpose
```

File: api/README.md [hunk 2]
```diff
@@ -122,16 +134,67 @@ try {
...
+const { codeHash } = gearApi.code.upload(code);
+gearApi.code.signAndSend(alice, () => {
+  events.forEach(({ event: { method, data } }) => {
+    if (method === 'ExtrinsicFailed') {
+      throw new Error(data.toString());
+    } else if (method === 'CodeSaved') {
+      console.log(data.toHuman());
+    }
+  });
+});
```
Comment (nikitayutanov) [inline api/README.md:157]:
```suggestion
const code = fs.readFileSync('path/to/program.opt.wasm');

const { codeHash } = gearApi.code.upload(code);

gearApi.code.signAndSend(alice, () => {
  events.forEach(({ event: { method, data } }) => {
    if (method === 'ExtrinsicFailed') {
      throw new Error(data.toString());
    } else if (method === 'CodeSaved') {
      console.log(data.toHuman());
    }
  });
});
```

File: api/README.md [hunk 3]
```diff
@@ -122,16 +134,67 @@ try {
...
+  gasLimit: 1000000,
+  value: 1000,
+  initPayload: somePayload,
+};
+
+try {
+  const { programId, salt, extrinsic } = gearApi.program.create(program, meta);
+} catch (error) {
+  console.error(`${error.name}: ${error.message}`);
+}
```
Comment (nikitayutanov) [inline api/README.md:180]: Do you think that we should mention `try-catch`?
Imo it's up to user and hurts readability.

File: api/README.md [hunk 4]
```diff
@@ -208,20 +264,29 @@ const paymentInfo = await api.program.paymentInfo(alice);
...
 ### Calculate gas for messages
+It's possible to find out the minimum gasLimit value to specify when sending extrinsic.
+The `gearApi.program.calculateGas.[method]` will help.
 
 Gas calculation returns GasInfo object contains 3 parameters:
 
 - `min_limit` - Minimum gas limit required for execution
 - `reserved` - Gas amount that will be reserved for some other on-chain interactions
 - `burned` - Number of gas burned during message processing
+- `may_be_returned` - value that can be returned in some cases
```
Comment (nikitayutanov) [inline api/README.md:280]: Prolly worth to specify what cases

File: api/README.md [hunk 5]
```diff
@@ -230,8 +295,28 @@ const gas = await gearApi.program.calculateGas.init(
...
+```javascript
+const codeId = '0x...';
+const gas = await gearApi.program.calculateGas.initCreate(
+  '0xd43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d', // source id
+  codeId,
+  '0x00', // payload
+  0, //value
+  true, // allow other panics
+);
+console.log(gas.toHuman());
```
Comment (nikitayutanov) [inline api/README.md:313]:
```suggestion
const codeId = '0x...';

const gas = await gearApi.program.calculateGas.initCreate(
  '0xd43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d', // source id
  codeId,
  '0x00', // payload
  0, //value
  true, // allow other panics
);

console.log(gas.toHuman());
```

File: api/README.md [hunk 6]
```diff
@@ -208,20 +264,29 @@ const paymentInfo = await api.program.paymentInfo(alice);
 const transactionFee = paymentInfo.partialFee.toNumber();
 consolg.log(transactionFee);
 ```
+</details>
+
+
 
 ### Calculate gas for messages
+It's possible to find out the minimum gasLimit value to specify when sending extrinsic.
+The `gearApi.program.calculateGas.[method]` will help.
```
Comment (nikitayutanov) [inline api/README.md:273]:
```suggestion
To find out the minimum gas amount to send extrinsic, use `gearApi.program.calculateGas[method]`.
```

File: api/README.md [hunk 7]
```diff
@@ -268,48 +356,84 @@ const gas = await gearApi.program.calculateGas.reply(
...
 console.log(gas.toHuman());
 ```
+</details>
+
+---
+
+## Work with programs and blockchain state
+
+### Check that the address belongs to some program
+If you need to know if an address belongs to a program or not, you can use the `api.program.exists` method.
```
Comment (nikitayutanov) [inline api/README.md:366]:
```suggestion
Find out if address belongs to a program using the `api.program.exists` method.
```

File: api/README.md [hunk 8]
```diff
@@ -268,48 +356,84 @@ const gas = await gearApi.program.calculateGas.reply(
...
+<details>
+<summary>Example</summary>
+
+```javascript
+const programId = '0x0000000000000000000000000000000000000000000000000000000000000000';
+const programExists = await api.program.exists(programId);
+console.log(`Program with address ${programId} ${programExists ? 'exists' : "doesn't exist"}`);
```
Comment (nikitayutanov) [inline api/README.md:373]:
```suggestion
const programId = '0x...';
const programExists = await api.program.exists(programId);

console.log(`Program with address ${programId} ${programExists ? 'exists' : "doesn't exist"}`);
```

Comment (nikitayutanov) [inline api/README.md:373]:
```suggestion
const programId = '0x...';
const programExists = await api.program.exists(programId);

console.log(`Program with address ${programId} ${programExists ? 'exists' : "doesn't exist"}`);
```

File: api/README.md [hunk 9]
```diff
@@ -399,94 +528,170 @@ const unsub = await gearApi.gearEvents.subscribeToNewBlocks((header) => {
...
 ```
+
+</details>
+
+### Sign data
+
+<details>
+<summary>Example</summary>
+
+1. Creating signature
```
Comment (nikitayutanov) [inline api/README.md:659]:
```suggestion
1. Create signature
```

File: api/README.md (hunk 4 above)
Comment (osipov-mit) [reply api/README.md:280]: It depends on contract. We cannot know it

**Comment by EugenWay** [comment]
Nice job. I'll update wiki

**Review by AndrePanin — APPROVED** [review]

**Pull request merged**